### PR TITLE
feat: added logging to inbox events

### PR DESCRIPTION
**Pull request by MadsDue**
## TODO

Please complete and remove this TODO section before entering PR ready for review state.

- [x] Title: Follow [this guide](https://github.com/Mech0z/GitHubGuidelines)
- [ ] Write tests
- [ ] Update documentation
- [ ] Update [event catalog](https://energinet.atlassian.net/wiki/spaces/D3/pages/555581556/Event+catalog) if there are relevant changes to integration events
- [ ] Inform team Outlaws if there are relevant C4 changes

### CODE REVIEW

**Review by BjarkeMeier — COMMENTED** [review]
Nice, thx for adding observability 🙇

**Review by ebbeknudsen — APPROVED** [review]

**Review by ebbeknudsen — APPROVED** [review]

**Review by ebbeknudsen — DISMISSED** [review]

File: source/dotnet/wholesale-api/WebApi/Startup.cs [hunk 1]
```diff
@@ -52,6 +52,7 @@ public Startup(IConfiguration configuration, IWebHostEnvironment environment)
     public void ConfigureServices(IServiceCollection serviceCollection)
     {
         serviceCollection.AddModules(Configuration);
+        serviceCollection.AddHttpContextAccessor();
```
Comment (dstenroejl) [inline source/dotnet/wholesale-api/WebApi/Startup.cs:55]: Why do we need to do this?

File: source/dotnet/wholesale-api/Events/Events.Application/Workers/ServiceBusWorker.cs [hunk 2]
```diff
@@ -60,28 +61,30 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
     {
         if (_serviceBusProcessor == null) throw new ArgumentNullException();
 
-        using (_logger.BeginScope(_loggingScope))
```
Comment (dstenroejl) [inline source/dotnet/wholesale-api/Events/Events.Application/Workers/ServiceBusWorker.cs:63]: Why do we need to refactor this?

File: source/dotnet/wholesale-api/Events/Events.Application/Workers/ServiceBusWorker.cs [hunk 3]
```diff
@@ -60,28 +61,30 @@ protected override async Task ExecuteAsync(CancellationToken stoppingToken)
...
+        await _serviceBusProcessor.StartProcessingAsync(stoppingToken).ConfigureAwait(false);
     }
 
     protected abstract Task ProcessAsync(ProcessMessageEventArgs arg);
 
     private static ServiceBusProcessor CreateServiceBusProcessor(ServiceBusClient serviceBusClient, IOptions<ServiceBusOptions> options)
     {
+        var serviceBusProcessorOptions = new ServiceBusProcessorOptions()
+        {
+            AutoCompleteMessages = false, // Default is true
```
Comment (dstenroejl) [inline source/dotnet/wholesale-api/Events/Events.Application/Workers/ServiceBusWorker.cs:77]: If we change this we must change the message handling, and I cannot see that change?

**Review by dstenroejl — COMMENTED** [review]
Added comments, but I guess it would be most efficient if we can talk :)

File: source/dotnet/wholesale-api/Events/Events.Application/Workers/ServiceBusWorker.cs [hunk 4]
```diff
@@ -45,61 +43,42 @@ public async ValueTask DisposeAsync()
...
-        }
+        await _serviceBusProcessor.StopProcessingAsync(cancellationToken).ConfigureAwait(false);
+        await base.StopAsync(cancellationToken).ConfigureAwait(false);
+        Logger.LogWarning("{Worker} has stopped", _serviceName);
     }
 
     protected override async Task ExecuteAsync(CancellationToken stoppingToken)
     {
-        if (_serviceBusProcessor == null) throw new ArgumentNullException();
+        if (_serviceBusProcessor == null)
```
Comment (dstenroejl) [inline source/dotnet/wholesale-api/Events/Events.Application/Workers/ServiceBusWorker.cs:53]: If we want to validate the value of a private readonly field, we should do it when the class is constructed.

**Review by dstenroejl — COMMENTED** [review]
Could we also fix the logger in the other class that inherits from ServiceBusWorker, so it uses the protected Logger?

File: source/dotnet/wholesale-api/Events/Events.Application/Workers/ServiceBusWorker.cs (hunk 4 above)
Comment (MadsDue) [reply source/dotnet/wholesale-api/Events/Events.Application/Workers/ServiceBusWorker.cs:53]: Good catch

**Review by dstenroejl — APPROVED** [review]
LGTM

**Pull request merged**